### ISSUE TITLE
Is it possible to train CosineSimilarityLoss without labels?

### ISSUE THREAD

**Issue by olastor** [UKPLab/sentence-transformers#326]
I'd like to rebuild a model from a certain paper that minimizes `1 - CosineSimilarity(a, b)`, but it seems to me that the implementation of CosineSimilarityLoss in this repo and the usage in the examples always require a label (i.e. -1 or 1). Is it possible to train the `CosineSimilarityLoss` loss without labels, providing only positive/similar example pairs?

**Comment by nreimers** [comment]
Hi @olastor 
The CosineSimilarityLoss requires a continous label, e.g. from -1 to 1 or from 0 to 1.

If you have only positive pairs, you can construct negative pairs by randomly selecting two pairs. This usually gives you a negative pair.
Then you can train with ConstrativeLoss.

Else have a look at triplet loss.

Best
Nils Reimers

**Comment by olastor** [comment]
@nreimers Thank you for the quick response and explanation!

**Comment by datistiquo** [comment]
> The CosineSimilarityLoss requires a continous label, e.g. from -1 to 1 or from 0 to 1.


> If you have only positive pairs, you can construct negative pairs by randomly selecting two pairs. This usually gives you a negative pair.

So, it is also possible if you just have binary labels 0,1 (and not continous) like for softmax or contrastive loss to train a CosineSimilarityLoss ? Otherwise it is actually difficult to get score like the gold scores in practice.

**Comment by nreimers** [comment]
Have a look at the different available loss functions:
https://www.sbert.net/docs/package_reference/losses.html

There are also some suitable if you just have binary labels like MultipleNegativesRankingLoss

**Comment by datistiquo** [comment]
Yes, just wondered about your statement that the labels have to be continous for Cosine Loss. But I think you could use it for just having binray labels too, right? Because this Loss is just a normal MSE Loss?

**Comment by nreimers** [comment]
I think if you have labels 0 and 1, then cosine sim schould also work.

**Comment by datistiquo** [comment]
> I think if you have labels 0 and 1, then cosine sim schould also work.

But this is eaxatcly what you also do in the paper, right? So I am a bit unsure what your "should work" means?

> Regression Objective Function. The cosinesimilarity between the two sentence embeddings
u and v is computed (Figure 2). We use meansquared-error loss as the objective function.
